test/test_torch.py

```diff
@@ -11128,6 +11128,19 @@ def test_sign(self, device):
         a_bool.sign_()
         self.assertEqual(a_bool, a_bool_target, msg='sign_ device={} dtype=bool'.format(device))
 
+    @dtypes(torch.cfloat, torch.cdouble)
+    def test_sgn(self, device, dtype):
+        x = torch.randn(100, dtype=dtype)
+        angle = x.angle()
+        cos_angle = angle.cos()
+        sin_angle = angle.sin()
+        expected = cos_angle + 1j * sin_angle
+        self.assertEqual(x.sgn(), expected)
```

Comment:
Can this instead assert that x.sgn has the same angle as x.angle and that the absolute value is every one?

Reply:
yeah sure